Makefile

```diff
@@ -163,6 +163,14 @@ python-image:
 push-python-image:
 	docker push ${IMAGE_PREFIX}python:${TEST_INFRA_VERSION}
 
+# Build the csharp build image
+csharp-image:
+	docker build -t ${INIT_IMAGE_PREFIX}build/csharp:${TEST_INFRA_VERSION} containers/init/build/csharp
+
+# Push the csharp build image to a docker registry
+push-csharp-image:
+	docker push ${INIT_IMAGE_PREFIX}build/csharp:${TEST_INFRA_VERSION}
```

Comment:
Similar to the configure script where we introduced a new variable for BUILD_IMAGE_PREFIX, I think we should do that in the Makefile. If I set $INIT_IMAGE_PREFIX="" and try to push to DockerHub registry, it would push as build/csharp which would not be valid. So, I'd like to move the build/ part to be customizable as well. Can we introduce a ${BUILD_IMAGE_PREFIX}? It would also require a change to deployment guide.

Reply:
If we have this BUILD_IMAGE_PREFIX, should we also ask BUILD_IMAGE_PREFIX to be set up during the deployment process? If we do, since build is part of the init containers we separating them with two environment variables could create some confusion. I agree we should make sure it is compatible with all registries, just feel the naming of the environment is a bit confusing.